src/common/deploy-geo.js

```diff
@@ -0,0 +1,51 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/*jshint forin:true, noarg:false, noempty:true, eqeqeq:true, bitwise:true,
+  strict:true, undef:true, curly:false, browser:true,
+  unused:true,
+  indent:2, maxerr:50, devel:true, node:true, boss:true, white:true,
+  globalstrict:true, nomen:false, newcap:true, esnext: true, moz: true  */
+
+/*global require, console, exports, geoip_country_code */
+
+// adapted from:
+
+"use strict";
+
+// bascially, if we think it's august 2015, do a special rollout, otherwise,
+// don't.
+
+const geo = require("./geo");
+const fakeUserCountry = () => Promise.resolve("--");
+
+
+`logic
+
+1.  is it august 2015?  If so, answer proporitional to day
+2.  else:  just answer, darn it
+`
+
+var useReal = function (now, rng) {
+  now = now || new Date();
+  if (rng === undefined) rng = Math.random();
+  var d = now.getDate(),
+      m = now.getMonth(),
+      y = now.getFullYear();
+
+  // false ONLY in 2015/Aug *and* 'too early'
+  let ans = !(y == 2015 && m == 7 && ((d / 30) < rng) );
```

Comment:
Putting the `(d / 30) < rng` in the notted section of the condition makes it harder to read, at least for me. I'd write this as the more verbose but clearer:

```js
if (y === 2015 && m === 7) {
  return rng < (d / 30);
} else {
  return true;
}
```